src/python/gem5/components/devices/gpus/amdgpu.py

```diff
@@ -0,0 +1,267 @@
+# Copyright (c) 2024 Advanced Micro Devices, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from this
+# software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from m5.objects import (
+    AMDGPUDevice,
+    SubSystem,
+)
+
+from ....components.memory.abstract_memory_system import AbstractMemorySystem
+from ....prebuilt.viper.gpu_cache_hierarchy import ViperGPUCacheHierarchy
+from .viper_shader import ViperShader
+
+
+class BaseViperGPU(SubSystem):
+    _base_pci_dev = 8
+    _gpu_count = 0
+    _my_id = 0
+
+    @classmethod
+    def next_pci_dev(cls):
+        cls._gpu_count += 1  # Use count for this particular type
+        return cls._base_pci_dev + cls._gpu_count - 1
+
+    @classmethod
+    def get_gpu_count(cls):
+        return cls._gpu_count
+
+    def __init__(self, gpu_memory: AbstractMemorySystem):
```

Comment:
Some day (definitely not before this release), we should make this constructor take a list of shaders, a cache hierarchy, and a memory system. This way the GPU is just like a board.